src/nvim/ex_getln.c

```diff
@@ -1911,7 +1911,7 @@ static int command_line_changed(CommandLineState *s)
     //       - Immediately undo the effects.
     State |= CMDPREVIEW;
     emsg_silent++;  // Block error reporting as the command may be incomplete
-    do_cmdline(ccline.cmdbuff, NULL, NULL, DOCMD_KEEPLINE|DOCMD_NOWAIT);
```

Comment:
Can't we keep this and instead add `State & CMDPREVIEW` as an alternative in the enclosing if-statement?

Reply:
I try to think of it as I reach room, now in class

Reply:
By keeping `DOCMD_NOWAIT`, it means that the if part in `ex_docmd.c` will not work, I want that the if part should work, but only `no_wait_return` should not be incremented, if I have to edit it, the condition in if will become so messy, and according to my personal opinion, I don't like to read a messy, complicated and big condition, so I put this like the way.

Other alternative which I think that was good, was making a separate if condition for `State & CMDPREVIEW` but according to @justinmk its also not too better, so this option was left, which I thought was acceptable.
What are the views on it?

Reply:
But this is a no-wait state, just like async and `<Cmd>` commands in cmdline mode. Just having this flag silently missing when it is expected is also messy.

Reply:
Okk, then its an issue, then making the condition edit is better.